client/src/pages/Register.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from "react";
+import { withStyles } from "@material-ui/core/styles";
+//import { Router } from "react-router-dom";
+
+import axios from "axios";
+
+const registerPageStyle = theme => ({
+  landingContainer: {
+    margin: theme.spacing.unit * 2
+  }
+});
+
+class Register extends Component {
+  state = {
+    userName: " ",
+    userEmail: " ",
+    userPassword: " ",
+    confirmPassword: " "
+
+  };
+
+
+  handleSubmit = async (e) => {
+    e.preventDefault();
+    const userData = {
+      name: this.state.userName,
+      email: this.state.userEmail,
+      password: this.state.userPassword,
+      confirmPassword: this.state.confirmPassword
+    }
+    //make http post request to send name,email and password to server
+    const { data } = await axios.post("/api/users", userData)
+    console.log("new user info:", data);
+    //clear input fields
+    this.setState({
+      userName: " ",
+      userEmail: " ",
+      userPassword: "",
+      confirmPassword: " "
+    })
+
+    //direct user to profile page
+    this.props.history.replace("/profile")
```

Comment:
You'll want to confirm that the response from server is positive before redirecting the user

Reply:
You probably also want to wrap the request in a try/catch block to handle cases where there's an error in the server